docs/source/explanations/license-detection-reference.rst

```diff
@@ -0,0 +1,194 @@
+  getting a unique detections across a scan referencing the details.
+
+- work on primary-license and license scoring.
+
+LicenseDetection
+----------------
+
+This aims to solve a few types of false positives commonly observed in
+ScanCode license detection. These are:
+
+The ``unknown`` cases
+^^^^^^^^^^^^^^^^^^^^^
+
+- Unknown Intros with Proper Detections after them
+- Unknown references to local files
+
+License Clues
+^^^^^^^^^^^^^
+
+Also this would introduce a ``license_clues`` list of LicenseMatches
+which would have improper detections or other clues like urls which
+cannot be marked as detections.
+
+License Versions
+^^^^^^^^^^^^^^^^
+
+This would also simplify license-expressions for gpl/lgpl cases
+with versioned/unversioned matches detected together.
+
+Package License Detections
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+License detections in package manifests now just have the license-expression
+from the detection and this is different from licenses detected directly which
+have details. So packages now would also have details.
+
+Other Soulution Elements
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Merged:
+
+- Key {{phrases}} in license text rules
+- New license clarity scoring
+- Report the primary license
+
+Upcoming:
+
+- Make it easier to report, review and curate license detections
+  (GSoC Project in scancode.io)
+
+- Fixing bugs and updating the heuristics.
+  (This will be ongoing like the LicenseDB updates)
+
+Examples
+^^^^^^^^
+
+An example from the eclipse foundation::
+
+ /*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+
+The text ``"This program and the accompanying materials are made\n* available under the terms
+of the",`` is detected as ``unknown-license-reference`` with ``is_license_intro`` as True,
+and has several ``"epl-2.0"`` detections after that.
+
+What is a LicenseDetection?
+---------------------------
+
+A detection which can have one or multiple LicenseMatch in them,
+and creates a License Expression that we finally report.
+
+Properties:
+
+- A file can have multiple LicenseDetections (seperated by non-legalese lines)
+- This can be from a file directly or a package.
+- We should be mostly certain of a proper detection to create a LicenseDetection.
+- One LicenseDetection can have matches from different files, in case of local license
+  references.
+
+
+LicenseMatch Result Data
+------------------------
+
+LicenseMatch data currently is based on a ``license key`` instead of being based
+on an ``license-expression``.
+
+So if there is a ``mit and apache-2.0`` license expression detected from a single
```

Comment:
Is case important with license expression operands? Does `AND` usually get capitalized?

Reply:
Usually no, we use upper case actually in our output license expressions but https://github.com/nexB/license-expression/ can parse these expression irrespective of upper/lower case.